sefaria/views.py

```diff
@@ -170,28 +170,62 @@ def accounts(request):
     })
 
 
-def subscribe(request, email):
+def generic_subscribe_to_newsletter_api(request, email, subscribe):
     """
-    API for subscribing to mailing lists, in `lists` url param.
-    Currently active lists are:
-    "Announcements_General", "Announcements_General_Hebrew", "Announcements_Edu", "Announcements_Edu_Hebrew"
+    Generic view for subscribing a user to a newsletter
     """
     body = json.loads(request.body)
-    language = body.get("language", "")
-    educator = body.get("educator", False)
     first_name = body.get("firstName", None)
     last_name = body.get("lastName", None)
     try:
-        crm_mediator = CrmMediator()
-        if crm_mediator.subscribe_to_lists(email, first_name, last_name, educator=educator, lang=language):
+        if subscribe(request, email, first_name, last_name):
             return jsonResponse({"status": "ok"})
         else:
-            logger.error("Failed to subscribe to list")
+            logger.error(f"Failed to subscribe to list")
             return jsonResponse({"error": _("Sorry, there was an error.")})
     except ValueError as e:
         logger.error(f"Failed to subscribe to list: {e}")
         return jsonResponse({"error": _("Sorry, there was an error.")})
 
+
+def subscribe_sefaria_newsletter_view(request, email):
+    return generic_subscribe_to_newsletter_api(request, email, subscribe_sefaria_newsletter)
+
+
+def subscribe_steinsaltz_newsletter_view(request, email):
+    return generic_subscribe_to_newsletter_api(request, email, subscribe_steinsaltz)
+
+
+def subscribe_sefaria_newsletter(request, email, first_name, last_name):
+    """
+    API for subscribing to mailing lists, in `lists` url param.
+    Currently active lists are:
+    "Announcements_General", "Announcements_General_Hebrew", "Announcements_Edu", "Announcements_Edu_Hebrew"
+    """
+    body = json.loads(request.body)
+    language = body.get("language", "")
+    educator = body.get("educator", False)
+    crm_mediator = CrmMediator()
+    return crm_mediator.subscribe_to_lists(email, first_name, last_name, educator=educator, lang=language)
```

Comment:
I'm not sure why this removes the (not so much) more informative logging + json response there was before?

Reply:
It doesn't, the logging now happens in `generic_subscribe_to_newsletter_api()`